### PR TITLE
deploy: Inject Jazzer version via a flag

### PR DESCRIPTION
**Pull request by fmeum**
This removes the need to have a "version bump" commit before every release.

### CODE REVIEW

**Comment by fmeum** [comment]
Successful run: https://github.com/CodeIntelligenceTesting/jazzer/actions/runs/6248994924

File: .github/workflows/prerelease.yaml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 name: Pre-Release
 
 on:
-  workflow_dispatch:
```
Comment (zgtm) [inline .github/workflows/prerelease.yaml:4]: Maybe it's safer to leave this in for the time being, for the cases where we might need it anyway? :grimacing:

File: .github/workflows/prerelease.yaml [hunk 2]
```diff
@@ -126,18 +130,9 @@ jobs:
...
-      - name: read version
-        id: read-version
-        run: |
-          echo ::set-output name=version::\
-          $(sed -nr 's/JAZZER_VERSION = "(.*)"/\1/p' maven.bzl)
-        shell: bash
-
       - name: create release
-        uses: softprops/action-gh-release@v1
+        uses: softprops/action-gh-release@c9b46fe7aad9f02afd89b12450b780f52dacfb2d
```
Comment (zgtm) [inline .github/workflows/prerelease.yaml:147]: This is just for stability reasons, I guess? Or is there something in the newer commits that we need?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -70,11 +70,8 @@ Run `./format.sh` to format all source files in the way enforced by the "Check f
...
 
-1. Update `JAZZER_VERSION` in [`maven.bzl`](maven.bzl).
-2. Trigger the "Prerelease" GitHub Actions workflow for the branch where you want to do the release:
-    * `main` for major and minor releases
-    * An appropriate release branch for patch releases
-3. Wait for the workflow to finish (about 10 minutes)
-4. When successful and happy with the results, log into https://oss.sonatype.org, select both staging repositories and "Close" them.
+1. Push a tag of the form `v1.2.3` to trigger the "Prerelease" GitHub Actions workflow.
+2. Wait for the workflow to finish (about 10 minutes)
+3. When successful and happy with the results, log into https://oss.sonatype.org, select both staging repositories and "Close" them.
```
Comment (zgtm) [inline CONTRIBUTING.md:75]: Just realized that I missed this in my change of the of the steps, but this should read "all three" instead of "both" :smile:

Comment (zgtm) [inline CONTRIBUTING.md:75]:
```suggestion
3. When successful and happy with the results, log into https://oss.sonatype.org, select all three staging repositories (under "Repositories" -> "Staging" -> `com/code_intelligence`) and "Close" them.
```

**Review by zgtm — APPROVED** [review]
Looks good! Makes our lives even easier! :+1:

**Comment by fmeum** [comment]
It worked: https://github.com/CodeIntelligenceTesting/jazzer/actions/runs/6249512046

**Pull request merged**